perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/pwdmgr/EinfraPasswordManagerModule.java

```diff
@@ -67,6 +67,27 @@ public EinfraPasswordManagerModule() {
 
 	}
 
+	@Override
+	public Map<String, String> generateAccount(PerunSession sess, Map<String, String> parameters) {
+		String password = parameters.get(PASSWORD_KEY);
+		String login = parameters.get(LOGIN_PREFIX + "einfra");
+
+		if (password == null || password.isBlank()) {
```

Comment:
Both login and password are checked within ``reservePassword()`` method and with better debug/descriptions. If you insist on using `generateAccount()` method for purpose of reservation, I would rather add exceptions to API, so that caller knows what happened, instead of wrapping them to generic `InternalErrorException`.

Reply:
This is also checked in the `checkSponsoredUserData` method, which is used in the `MembersManagerBl`, and there is a special Exception thrown, with a more reasonable message. This was meant to be an internal check, more than a validation of the user's input.